### ISSUE TITLE
Review new Knack pipeline proof-of-concept

### ISSUE THREAD

**Issue by johnclary** [cityofaustin/atd-data-tech#3785]
I've started building a new Knack ETL pipeline, [atd-knack-services](https://github.com/cityofaustin/atd-knack-services).

I've done just enough to prove the concept and would like to get feedback from the dev team before going any further. 

Goals of this new design:
- reduce knack server load by using an S3 data store for common/frequent tasks
- keep the code as modular as possible
- make the code easy to maintain
- use Knackpy 1.0, which abstracts some of the headaches when writing Knack ETLs
- ability to run the new Knack ETLs on Airflow
- lay the groundwork for scaling to integrate with other services, e.g. a data warehouse or a query service like Athena

Open questions:
- does this approach make sense? i'm open to going in a different direction.
- does the repo structure make sense? do scripts belong in a separate repo?
- does the S3 file system structure make sense?

TODO's
- minor refactoring to make proper use env vars (holding off on this until I better understand the Airflow dev environment)
- write tests
- create DAGs, test with Airflow

**Comment by sergiogcx** [comment]
Question for @johnclary do knack datasets have a sequential primary key? Or is there a way to implement one? Reason I am asking is because this could help with integrations, instead of dumping entire datasets we can work the integrations to be progressive.

We would still have the full dataset in S3, but we would also save money in S3 (bandwidth costs) and it would be less resource expensive on Knack.

**Comment by johnclary** [comment]
@sergiogcx yes. we can add an auto-incrementing integer to any dataset. the current design uses created/modified timestamps to select which records to process. but you're suggesting a sequence would have other benefits?

**Comment by amenity** [comment]
@johnclary - do you want to schedule a discussion for this?
